### PR TITLE
Feature/locale page

### LINKED ISSUE #898: Figure out multilanguage sites
We have a way to tag language for values, but don't yet have a system for using them, other than the signal about what language a value has set. Seems like some way to 1) set a language, or languages to display and 2) filter based on that setting might be in order. Alternatively, people have suggested a way to have multiple sites that use different languages (which sounds like a site setting). Either of those approaches sound like having the theme be able to know what language(s) to display and filter values accordingly.

One feature we know people want is on-the-fly language switching, without being logged in and without changing browser setting. However, a system that _can_ work with browser settings also makes sense for the filtering.

See convo on the forums https://forum.omeka.org/t/internationalized-omeka-s/3154

### PR DESCRIPTION
**Pull request by Daniel-KM**
This feature is the required first step to make Omeka full multilingual, as many users want. 

There are two main ways to manage localization: one site by language, or multilanguage by site. The first way has many limitation, in particular it implies to have two urls for the same pages and documents. The second way is the normal way we found on many cms. 

Because there is a new value in the database for the site pages, it can be used by future modules easily, unlike in Omeka Classic.

Fix #898, #1293, https://forum.omeka.org/t/bilingual-content-internationalisation/2116, https://forum.omeka.org/t/internationalized-omeka-s/3154/22, etc.

### CODE REVIEW

**Comment by Daniel-KM** [comment]
For  #1349 (1.3) ?

**Comment by zerocrates** [comment]
It won't go into 1.3.

I'm not sure about this in general as the solution though... you note the differences between the basic strategies for doing multilingual within the sites, but I'm not sure language tagging at the page level gets us all that far... you'd still have basic problems with things like the slug of the site, the title of the site (also the slug of the pages which would still have to vary between different language "versions" of the same page, the new "summary" of the site, the strings in the navigation, the actual resource browse/show views, and the theme settings (things like text).

Obviously you've framed this as a first step so it's not exactly a problem that it doesn't deal with those things but my concern is it's a first step down a road of adding significant complexity to all those areas (at least). The main advantage of the "site" level of language tagging that we currently have is that all those pieces of simple data within the site are covered: you can still only have one slug, one title, and so on, but since the site can declare its language that's fine. Of course this implies duplicating the site to have versions of it in different languages, but you're fundamentally doing that duplication no matter the model.

In my experience different URLs for different languages of the same content is pretty standard, but I'd be happy to hear about counterexamples or alternatives.

In short: I'm not against language tagging at the page level like you're doing here, but I'm just not sure how it fits into an eventual "complete" multilingual solution for S sites.

**Comment by Daniel-KM** [comment]
Ok if you prefer the solution of one language by site, it's good to know, I'll fix my pr in that way. 
So this solution implies at least the possibility to have a "/" in the slug name in order to do `/my-site/fr` or `/my-site/en`, and/or in the page slug. This point refers too to the more general issue of web site management (use a domain by site, to have a main site).
